pkg/kv/kvserver/uncertainty/doc.go

```diff
@@ -192,6 +194,116 @@ var D6 = roachpb.Transaction{}.ObservedTimestamps
 
 // D7 ————————————————————————————————————————————————
 //
+// TimestampFromServerClock
+//
+// Non-transactional requests that defer their timestamp allocation to the
+// leaseholder of their (single) range also have uncertainty intervals, which
+// ensures that they also guarantee single-key linearizability even with only
+// loose (but bounded) clock synchronization. Non-transactional requests that
+// use a client-provided timestamp do not have uncertainty intervals and do not
+// make real-time ordering guarantees.
+//
+// Unlike transactions, which establish an uncertainty interval on their
+// coordinator node during initialization, non-transactional requests receive
+// uncertainty intervals from their target leaseholder, using a clock reading
```

Comment:
Since this is a choice, it makes sense to motivate it: we have all reason to believe that this means the leaseholder is chosing the timestamp, which means we get the a relevant observed timestamp "for free".